lib/rules/id-blacklist.js

```diff
@@ -141,6 +163,25 @@ module.exports = {
                         if (isInvalid(name)) {
                             report(node);
                         }
+
+                    // Report the last identifier in an ObjectPattern destructuring.
+                    } else if (
+                        (
+                            effectiveParent.type === "Property" &&
+                            effectiveParent.value === node.parent &&
+                            effectiveParent.parent.type === "ObjectPattern"
+                        ) ||
+                        effectiveParent.type === "RestElement" ||
+                        effectiveParent.type === "ArrayPattern" ||
+                        (
+                            effectiveParent.type === "AssignmentPattern" &&
+                            effectiveParent.left.type === "MemberExpression" &&
```

Comment:
I think that `effectiveParent.left.type === "MemberExpression" &&` isn't necessary here, it certainly is `MemberExpression` in this path?